llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp

```diff
@@ -851,6 +853,24 @@ LegalizerHelper::libcall(MachineInstr &MI, LostDebugLocObserver &LocObserver) {
       return Status;
     break;
   }
+  case TargetOpcode::G_FPOWI: {
+    Type *HLTy = getFloatTypeForLLT(Ctx, LLTy);
+    Type *ITy = IntegerType::get(
+        Ctx, MRI.getType(MI.getOperand(2).getReg()).getSizeInBits());
```

Comment:
You can `cast<GenericMachineInstr>(MI)` and then use `getReg(N)` for brevity here and below as well.